eden/scm/setup.py

```diff
@@ -372,9 +372,7 @@ def localhgenv():
         env["SystemRoot"] = os.environ["SystemRoot"]
     return env
 
-
```

Comment:
Prescribing a VCS without first detecting the repo type (hg, sl, git) still bugs me out. Could something be done for detecting the repo type?

Also, users having both hg and sl installed might be more common than one might think.

Reply:
I think this is an improvement. Previously `setup.py` hardcodes `"hg"` which obviously won't work in all cases. Whether we detect `.hg` or `.sl` or just try out both of them and pick one that works seems to be just an implementation detail to achieve the same goal, and I don't think that should block this PR.